fitsio/tests/test_vector_datatypes.rs

```diff
@@ -0,0 +1,111 @@
+use fitsio::tables::{ColumnDataType, ColumnDescription};
+use fitsio::FitsFile;
+use tempfile::Builder;
+
+const NV: usize = 128; // vector size
+const NR: usize = 3; // number of rows
+
+macro_rules! make_test {
+    ($func:ident, $t:ty, $dt:expr, $val:expr) => {
+        #[test]
+        fn $func() {
+            let tmp_dir = Builder::new().prefix("fitsio-").tempdir().unwrap();
+            let file_path = tmp_dir.path().join("example.fits");
+
+            let orig_data: Vec<$t> = vec![$val].repeat(NV * NR);
+            {
+                let mut fitsfile = FitsFile::create(&file_path).open().unwrap();
+
+                let col = ColumnDescription::new("TEST")
+                    .with_type($dt)
+                    .that_repeats(NV)
+                    .create()
+                    .unwrap();
+                let columns = &[col];
+                let table_hdu = fitsfile.create_table("DATA", columns).unwrap();
+                table_hdu
+                    .write_col(&mut fitsfile, "TEST", &orig_data)
+                    .unwrap();
+            }
+
+            let mut f = FitsFile::open(file_path).unwrap();
+
+            let table_hdu = f.hdu("DATA").unwrap();
+            let data: Vec<$t> = table_hdu.read_col(&mut f, "TEST").unwrap();
+            assert!(orig_data.iter().zip(&data).all(|(a, b)| a == b));
```

Comment:
Again can we double check the length of the returned vector

```suggestion
            assert_eq!(orig_data.len(), data.len());
            assert!(orig_data.iter().zip(&data).all(|(a, b)| a == b));
```